player/client.c

```diff
@@ -1591,35 +1594,40 @@ static bool gen_property_change_event(struct mpv_handle *ctx)
     ctx->lowest_changed = ctx->num_properties;
     for (int n = start; n < ctx->num_properties; n++) {
         struct observe_property *prop = ctx->properties[n];
-        if ((prop->changed || prop->updating) && n < ctx->lowest_changed)
+        if ((prop->changed || prop->updating || prop->updated) && n < ctx->lowest_changed)
```

Comment:
This is to check properties whose value is being retrieved, so `(prop->changed || prop->updating || prop->updated)` can be simplified to `(prop->need_new_value || prop->updating)`, as event can be generated immediately in other cases.